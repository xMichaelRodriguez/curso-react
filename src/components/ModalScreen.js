import React, { useState } from "react";
import LoadingMask from "react-loadingmask";
import "react-loadingmask/dist/react-loadingmask.css";
import Modal from "react-modal";
import { useDispatch, useSelector } from "react-redux";
import { startDeletedImage } from "../actions/images";
import { uiCloseModal } from "../actions/uiActions";

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    padding: "0",
    transform: "translate(-50%, -50%)",
  },
};

// Make sure to bind modal to your appElement (https://reactcommunity.org/react-modal/accessibility/)
Modal.setAppElement("#root");

export const ModalSchreen = () => {
  const dispatch = useDispatch();
  const [Loading, setLoading] = useState(false);
  const [deleted, setDeleted] = useState("");
  const { modalOpen } = useSelector((state) => state);
  let subtitle;

  const afterOpenModal = () => {
    // references are now sync'd and can be accessed.
    subtitle.style.color = "#fffff";
  };

  const closeModal = () => {
    dispatch(uiCloseModal());
  };

  const handleDelete = async () => {
    setLoading(true);
    const resp = await dispatch(startDeletedImage());
    if (resp.ok) {
      setDeleted(resp.msg);
      closeModal();
    }

    setLoading(false);
  };
  return (
    <Modal
      isOpen={modalOpen}
      onAfterOpen={afterOpenModal}
      onRequestClose={closeModal}
      style={customStyles}
      contentLabel="Delete Modal"
      overlayClassName="modal-fondo"
    >
      <div className="card">
        <div className="card-header">
          <button
            onClick={closeModal}
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
          <h2
            className="card-title"
            ref={(_subtitle) => (subtitle = _subtitle)}
          >
            Eliminacion de Imagen
          </h2>
        </div>
        <div className="card-body">
          {!!deleted ? (
            <h3>{deleted}</h3>
          ) : Loading ? (
            <div>
              <LoadingMask
                style={{ backgroundColor: "#999999", heigth: "100%" }}
                loading={Loading}
                text={"loading..."}
              >
                <h3>Estas seguro de eliminarla?</h3>
              </LoadingMask>
            </div>
          ) : (
            <h5>Estas seguro de eliminarla?</h5>
          )}
        </div>
        <div className="card-footer d-flex justify-content-between">
          <button
            type="button"
            className="btn btn-primary"
            onClick={handleDelete}
          >
            Si, borrala
          </button>
          <button
            type="button"
            className="btn btn-secondary"
            onClick={closeModal}
          >
            cerrar
          </button>
        </div>
      </div>
    </Modal>
  );
};
