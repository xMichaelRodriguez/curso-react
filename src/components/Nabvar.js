import React from "react";

export const Nabvar = () => {
  return (
    <div>
      <nav className="navbar navbar-light bg-dark">
        <div className="container-fluid d-flex justify-content-center">
          <div className="input-group ">
            <form className="form-inline my-2 my-md-0 w-100 d-flex justify-content-center">
              <input
                className="form-control mr-sm-2 w-75"
                type="text"
                placeholder="Search"
              />
              <button className="btn btn-secondary  rounded-circle my-2 my-sm-0" type="button" >
              <i className="fas fa-search"></i>
              </button>
            </form>
          </div>
        </div>
      </nav>
    </div>
  );
};
