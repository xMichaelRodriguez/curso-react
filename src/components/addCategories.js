import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategories = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const inputChange = (e) => {
    setInputValue(e.target.value);

    console.log("hola me llamaste?")
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cat) => [inputValue, ...cat]);
     setCategories('')
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <div className="row">
        <div className="input-field col s12">
          <input type="text" value={inputValue}  onChange={inputChange} />
          <label forof="text">Name of gift</label>
        </div>
      </div>
    </form>
  );
};

AddCategories.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
