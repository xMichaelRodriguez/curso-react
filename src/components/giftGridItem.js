import React from "react";

export const GiftGridItem = ({ id, title, url }) => {
  return (
    <div className="col s12 m6 l6">
      <div className="card m6 z-depth-3">
        <div className="card-image ">
          <img src={url} alt={title} height="150px" width="150px" />
          <span className="card-title">{title}</span>
        </div>
      </div>
    </div>
  );
};
