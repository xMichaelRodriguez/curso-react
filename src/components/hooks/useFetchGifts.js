import { useEffect, useState } from "react";
import { getGift } from "../helpers/getGfits";

export const useFetchGifts = (categories) => {
  const [state, SetState] = useState({ data: [], loading: true });

  useEffect(() => {
    getGift(categories).then((img) => {
      
        SetState({
          data: img,
          loading: false,
        });
     
    });
  }, [categories]);
  return state;
};
