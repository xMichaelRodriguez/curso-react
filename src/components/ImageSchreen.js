import React from "react";
import { useDispatch } from "react-redux";
import { uiOpenModal } from "../actions/uiActions";
import { ModalSchreen } from "./ModalScreen";
export const ImageSchreen = ({ image, id }) => {
  const dispatch = useDispatch();

  const handleImageClick = () => {
    dispatch(uiOpenModal());
  };
  return (
    <>
      <div className="card shadow text-white">
        <img
          className="card-img-top"
          onDoubleClick={handleImageClick}
          src={image}
          key={id}
          alt="...."
        ></img>
      </div>

      <ModalSchreen item={id} />
    </>
  );
};
