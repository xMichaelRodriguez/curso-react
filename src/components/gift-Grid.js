import React from "react";
import { GiftGridItem } from "./giftGridItem";
import { useFetchGifts } from "./hooks/useFetchGifts";

export const GiftGrid = ({ categories }) => {
  const { data: images, loading } = useFetchGifts(categories);

  return (
    <>
      <h3>{categories}</h3>

      {loading && <p className=" animate__animated animate__flash">Loading...</p>}
      <div className="container">
        <div className="row ">
          {images.map((img) => (
            <GiftGridItem key={img.id} {...img} />
          ))}
        </div>
      </div>
    </>
  );
};
