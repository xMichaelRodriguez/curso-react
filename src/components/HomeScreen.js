import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { startImageLoading } from "../actions/images";
import { ImageSchreen } from "./ImageSchreen";

export const HomeScreen = () => {
  const dispatch = useDispatch();
  const imagesStore = useSelector((state) => state.images);

  useEffect(() => {
    dispatch(startImageLoading(imagesStore));
  }, [dispatch]);

  return (
    <>
      <div className="row">
        <div className="card-columns">
          {imagesStore !== "" &&
            imagesStore.map(({ image, id }) => (
              <ImageSchreen image={image} id={id} key={id} />
            ))}
        </div>
      </div>

      <Link to="/newFile" className="fabs btn btn-info rounded-circle ">
        <i className="fas fa-plus fa-3x"></i>
      </Link>
    </>
  );
};
