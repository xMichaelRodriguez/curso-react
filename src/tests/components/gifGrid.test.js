import React from "react";

import { shallow } from "enzyme";
import { GiftGrid } from "../../components/gift-Grid";
import { useFetchGifts } from "../../components/hooks/useFetchGifts";
import { mockReturn } from "react-dom/cjs/react-dom-test-utils.development";
jest.mock("../../components/hooks/useFetchGifts");

describe("Debe Cargar el component <GiftGrid/>", () => {
  const categories = "naturo";
  test("el component <GiftGrid/> ", () => {
    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GiftGrid categories={categories} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items al cargar imagenes", () => {
    const gift = {
      id: "ABC",
      title: "asd",
      url: "https://localhost/algo/algo.png",
    };
    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GiftGrid categories={categories} />);

    expect(wrapper).toMatchSnapshot()
  });
});
