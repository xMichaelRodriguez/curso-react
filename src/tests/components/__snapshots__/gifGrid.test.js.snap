// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe Cargar el component <GiftGrid/> debe de mostrar items al cargar imagenes 1`] = `
<Fragment>
  <h3>
    naturo
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="container"
  >
    <div
      className="row "
    >
      <div
        className="col s12 m8 l6"
      >
        <div
          className="card m6 z-depth-3 "
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`Debe Cargar el component <GiftGrid/> el component <GiftGrid/>  1`] = `
<Fragment>
  <h3>
    naturo
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="container"
  >
    <div
      className="row "
    >
      <div
        className="col s12 m8 l6"
      >
        <div
          className="card m6 z-depth-3 "
        />
      </div>
    </div>
  </div>
</Fragment>
`;
